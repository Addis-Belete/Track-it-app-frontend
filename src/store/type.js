export const FETCH_PROGRESS_REQUEST = 'FETCH_PROGRESS_REQUEST';
export const FETCH_PROGRESS_SUCCESS = 'FETCH_PROGRESS_SUCCESS';
export const FETCH_PROGRESS_FAILURE = 'FETCH_PROGRESS_FAILURE';
