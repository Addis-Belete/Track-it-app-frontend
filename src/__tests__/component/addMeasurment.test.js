import React from 'react';
import { cleanup, render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddMeasurement from '../../component/AddMeasurment';
import store from '../../store/store';
import '@testing-library/jest-dom';

afterEach(cleanup);
describe('addMeasurment', () => {
  it('should render form to add measurment', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddMeasurement />
        </BrowserRouter>

      </Provider>,

    );
    expect(
      asFragment(
        <Provider store={store}>
          <AddMeasurement />
        </Provider>,

      ),
    );
  });
});
