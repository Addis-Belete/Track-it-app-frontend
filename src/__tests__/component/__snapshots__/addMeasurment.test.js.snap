// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMeasurment should render form to add measurment 1`] = `
<DocumentFragment>
  <div>
    <form
      class="form"
    >
      <select
        class="select"
        id="cat"
        name="id"
      >
        <option
          value="1"
        >
          Left bicep
        </option>
        <option
          value="2"
        >
          Right bicep
        </option>
        <option
          value="3"
        >
          Waist
        </option>
        <option
          value="4"
        >
          Hips
        </option>
        <option
          value="5"
        >
          Left thigh
        </option>
        <option
          value="6"
        >
          Right thigh
        </option>
      </select>
      <input
        class="input"
        max="100"
        min="0"
        name="result"
        placeholder="0.01CM"
        step="0.1"
        type="number"
        value=""
      />
      <button
        class="button"
        type="submit"
      >
        Add Measurment
      </button>
    </form>
    <div
      class="footer"
    >
      <a
        href="/addMeasurment"
      >
        <div
          class="add"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus-square fa-w-14 "
            data-icon="plus-square"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              fill="currentColor"
            />
          </svg>
          <p>
            Add Measure
          </p>
        </div>
      </a>
      <a
        href="/track"
      >
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chart-line fa-w-16 "
            data-icon="chart-line"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
              fill="currentColor"
            />
          </svg>
          <p>
            Track
          </p>
        </div>
      </a>
      <a
        href="/Progress"
      >
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chart-pie fa-w-17 "
            data-icon="chart-pie"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 544 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M527.79 288H290.5l158.03 158.03c6.04 6.04 15.98 6.53 22.19.68 38.7-36.46 65.32-85.61 73.13-140.86 1.34-9.46-6.51-17.85-16.06-17.85zm-15.83-64.8C503.72 103.74 408.26 8.28 288.8.04 279.68-.59 272 7.1 272 16.24V240h223.77c9.14 0 16.82-7.68 16.19-16.8zM224 288V50.71c0-9.55-8.39-17.4-17.84-16.06C86.99 51.49-4.1 155.6.14 280.37 4.5 408.51 114.83 513.59 243.03 511.98c50.4-.63 96.97-16.87 135.26-44.03 7.9-5.6 8.42-17.23 1.57-24.08L224 288z"
              fill="currentColor"
            />
          </svg>
          <p>
            Progress
          </p>
        </div>
      </a>
      <a
        href="/"
      >
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-out-alt fa-w-16 "
            data-icon="sign-out-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              fill="currentColor"
            />
          </svg>
          <p>
            Logout
          </p>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
